### PR TITLE
Fix timeout check workflow lint job

### LINKED ISSUE #142485: UNSTABLE Lint / workflow-checks / linux-job
The jobs cancels as it times out when running:

```
 python3 /home/ec2-user/actions-runner/_work/pytorch/pytorch/test-infra/.github/scripts/run_with_env_secrets.py ''
```

cc @seemethere @malfet @pytorch/pytorch-dev-infra @huydhn

### PR DESCRIPTION
**Pull request by huydhn**
Fixes https://github.com/pytorch/pytorch/issues/142485

The workflow check lint job timed out in trunk, i.e. https://github.com/pytorch/pytorch/actions/runs/12261226178/job/34207762939, and here was what happened:

1. https://github.com/pytorch/pytorch/pull/142294 landed yesterday to build ROCm on 3.13, but the PR had a landrace with https://github.com/pytorch/pytorch/pull/142282 in the generated workflow file
2. The trunk lint check caught that in https://github.com/pytorch/pytorch/blob/main/.github/scripts/report_git_status.sh#L2
3. However, the script also attempted to print the difference with `git diff .github/workflows`.  This command was the one that stuck because `git diff` uses page by default and requires a prompt to display the next page ¯\_(ツ)_/¯

It took so long to debug this because a timeout Nova GHA doesn't print any progress.  I'll create an issue for this.

Bonus:

I also fix the broken print from test tool lint job that confuses GitHub https://github.com/pytorch/pytorch/actions/runs/12261226178 with an annotation failure `Credentials could not be loaded, please check your action inputs`

### CODE REVIEW

File: .github/workflows/generated-linux-binary-manywheel-nightly.yml
Comment (wdvr) [inline .github/workflows/generated-linux-binary-manywheel-nightly.yml:1]: why is the generated file changed, but not the source file? Did we forget to generate on a previous commit?

**Review by wdvr — APPROVED** [review]

File: .github/workflows/generated-linux-binary-manywheel-nightly.yml
Comment (huydhn) [reply .github/workflows/generated-linux-binary-manywheel-nightly.yml:1]: It's a landrace https://github.com/pytorch/pytorch/pull/142282 removed some while https://github.com/pytorch/pytorch/pull/142294 l added them back.  Thus, rerunning the script will generate a new workflow file.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -207,8 +207,8 @@ jobs:
         conda activate "${CONDA_ENV}"
 
         # Test tools
-        python3 -m unittest discover -vs tools/test -p 'test_*.py'
-        python3 -m unittest discover -vs .github/scripts -p 'test_*.py'
+        PYTHONPATH=$(pwd) pytest tools/test/test_*.py
+        PYTHONPATH=$(pwd) pytest .github/scripts/test_*.py
```
Comment (ZainRizvi) [inline .github/workflows/lint.yml:211]: Isn't this only setting they PYTHONPATH env var,  resulting in no test files being executed?

Comment (huydhn) [reply .github/workflows/lint.yml:211]: Good catch!  The test works but one test there is not working with pytest https://github.com/pytorch/pytorch/actions/runs/12264094674/job/34217191327?pr=142476#step:14:135, so let me fix it

**Pull request closed without merging**